python/tvm/relay/frontend/pytorch_neo.py

```diff
@@ -0,0 +1,1019 @@
+            end = shape[:]
+            begin[axis] = i
+            end[axis] = i + unif_size
+            stride = [1] * len(shape)
+
+            chunk_out = _op.transform.strided_slice(data, begin, end, stride)
+            chunks.append(chunk_out)
+
+
+        if dim % num_chunks:
+            begin = [0] * len(shape)
+            end = shape[:]
+            begin[axis] = unif_size * (num_chunks - 1)
+            end[axis] = dim
+            stride = [1] * len(shape)
+
+            chunk_out = _op.transform.strided_slice(data, begin, end, stride)
+            chunks.append(chunk_out)
+
+        return chunks
+    return _impl
+
+def _matmul():
+    def _impl(inputs):
+        data0 = inputs[0]
+        data1 = inputs[1]
+        data1_t = _op.transpose(data1, axes=(1, 0))
+
+        return _op.nn.dense(data0, data1_t)
+    return _impl
+
+def _expand():
+    def _impl(inputs):
+        data_in = inputs[0]
+        if isinstance(data_in, _expr.Var):
+            shape = get_tensor_from_relay_var(data_in)
+        elif isinstance(data_in, (_expr.Call, _expr.TupleGetItem)):
+            shape = _infer_type(data_in).checked_type.shape
+
+        ndims = len(shape)
+        sizes = get_tensor_from_relay_var(inputs[1])
+        out = inputs[0]
+
+        for i in range(ndims):
+            if sizes[i] in {-1, shape[i]}:
+                continue
+            data = list()
+            for temp in range(sizes[i]):
+                data.append(out)
+            call = _op.tensor.concatenate(data, i)
+
+        return call
+    return _impl
+
+def _int():
+    def _impl(inputs):
+        if isinstance(inputs[0], _expr.Call):
+            return inputs[0]
+        return int(inputs[0])
+    return _impl
+
+def _listunpack():
+    def _impl(inputs):
+        return inputs[0]
+    return _impl
+
+def _to():
+    def _impl(inputs):
+        return inputs[0]
+    return _impl
+
+def _device():
+    def _impl(inputs):
+        return None
+    return _impl
+
+def _pad():
+    def _impl(inputs):
+        data = inputs[0]
+        padding = inputs[1]
+        pad_width = list(zip(padding, padding))
+        pad_value = inputs[2]
+        return _op.nn.pad(data, pad_width, pad_value)
+    return _impl
+
+def _sqrt():
+    def _impl(inputs):
+        data = inputs[0]
+        return _op.tensor.sqrt(data)
+    return _impl
+
+# Helper functions for operator implementation
+
+# Helper to grab actual tensor from a converted relay var (not sure if there's another way to do it)
+def get_tensor_from_relay_var(relay_var):
+    """ Get tensor by parsing the string of a relay var """
+    # v0.0.4
+    # free_var %v4: Tensor[(1, 1), float32]
+    # %v4
+
+    if 'Tensor' not in str(relay_var):
+        return []
+    temp = str(relay_var).split('Tensor[')[1] #(1, 1), float32]
+    tuple_str = temp.split(', float32')[0] #(1, 1) , float32]
+    tuple_str = tuple_str[1:-1] #1, 1
+
+    tensor = []
+    for i in tuple_str.split(', '):
+        tensor.append(int(i))
```

Comment:
Is it possible that `i` isn't an integer? for dynamic models, we need to use `Any` in relay to represent the size of a dimension here

Reply:
This is hardcoded so that would probably cause an error somewhere. I did remove this helper so infer_shape is used everywhere instead though.

Reply:
This is hardcoded so that would probably cause an error somewhere. I did remove this helper so infer_shape is used everywhere instead though.